static/app/components/forms/field/fieldControlState.tsx

```diff
@@ -9,11 +9,14 @@ type FieldControlStateProps = Pick<FieldGroupProps, 'flexibleControlStateSize'>;
 const FieldControlState = styled('div')<FieldControlStateProps>`
   display: flex;
   position: relative;
-  ${p => !p.flexibleControlStateSize && `width: 24px`};
   flex-shrink: 0;
   justify-content: end;
   align-items: center;
-  margin-left: ${space(0.5)};
+
+  ${p =>
+    p.flexibleControlStateSize
+      ? `&:not(:empty) { margin-left: ${space(1.5)} }`
```

Comment:
If `flexibleControlStateSize` is true and there is no control state inside, the wrapper shouldn't take up any width. We only need the margin when there is something inside.